### ISSUE TITLE
manifest: environment variable resolving using key only seems to be broken

### ISSUE THREAD

**Issue by nirgilboa** [okteto/okteto#3145]
**Describe the bug**
My manifest contains the following `environment` section with an empty key named `GITHUB_NPM_TOKEN`
```
    environment:
      GITHUB_NPM_TOKEN:
```
As described in the [manifest docs ](https://www.okteto.com/docs/reference/manifest/#environment-string-optional) using a key (without a value) should resolve to the value of the machine on which okteto is running. After successfully running `okteto up` and getting a shell on my container, the actual result is that the environment variable is resolved to an empty string
```
➜ env | grep GITHUB
GITHUB_NPM_TOKEN=
```

 Using `$` to pass in the value explicitly works:
```
    environment:
      GITHUB_NPM_TOKEN: $GITHUB_NPM_TOKEN
```
On the plus side, this method is more explicit, so a possible solution is to allow this method only.

**To Reproduce**

1. Define a manifest including an environment variable, without specifying its value (see example above)
2. Verify the environment variable has a value on the host machine
3. Run `okteto up` and open a shell on the container
4. Examine the value of the environment variable

**Expected behavior**
The environment variable defined without a value should resolve to its value on the host machine.

**Desktop:**

- OS: macOS 12.5.1
- Version: `okteto version 2.7.0`

**Comment by ifbyol** [comment]
@nirgilboa Thanks for reporting it. We were able to reproduce it and we'll review it.

As an alternative way, you should be able to specify the environment as a list of strings:

```
environment:
  - VAR1=value1
  - VAR2=value2
```

In that scenario, you can set

```
environment:
  - GITHUB_NPM_TOKEN
```

and it works.

**Comment by jmacelroy** [comment]
@jLopezbarb I know you hit some internal constraints on this in regards to libraries or parser. Do I have that correct? Should we update our docs to reflect that we don't support this yet and then figure out how we can support it? cc @omnomagonz

**Comment by jLopezbarb** [comment]
Yes, this is a known issue. Unfortunately it's not easy to solve because the problem is in the library that we use to parse the manifest `gopkg.in/yaml.v2`. It does not allow to parse a map structure with an empty value.

There is a new version `gopkg.in/yaml.v3` but that doesn't solve the issue about parsing maps with empty values
